api/controllers/MerchStoreController.ts

```diff
@@ -177,6 +179,21 @@ export class MerchStoreController {
     return { error: null, order };
   }
 
+  @Post('/order/verification')
+  async verifyMerchOrder(@Body() verifyOrderRequest: VerifyMerchOrderRequest,
+    @AuthenticatedUser() user: UserModel): Promise<ApiResponse> {
+    if (!PermissionsService.canAccessMerchStore(user)) throw new ForbiddenError();
+    const originalOrder = verifyOrderRequest.order.filter((oi) => oi.quantity > 0);
+    const orderIsEmpty = originalOrder.reduce((x, n) => x + n.quantity, 0) === 0;
+    if (orderIsEmpty) throw new UserError('There are no items in this order');
+    const numUniqueUuids = (new Set(originalOrder.map((oi) => oi.option))).size;
+    if (originalOrder.length !== numUniqueUuids) throw new BadRequestError('There are duplicate items in this order');
+
+    await this.merchStoreService.verifyOrder(originalOrder, user);
```

Comment:
nit: you can probably create a private function `validateMerchOrderRequest` or something similar that takes in an order and does this validation. `placeOrder()` can use this method as well so that there isn't logic duplication happening